controllers/workspace/devworkspace_controller.go

```diff
@@ -491,7 +491,7 @@ func (r *DevWorkspaceReconciler) doStop(workspace *dw.DevWorkspace, logger logr.
 func (r *DevWorkspaceReconciler) failWorkspace(workspace *dw.DevWorkspace, msg string, logger logr.Logger, status *currentStatus) (reconcile.Result, error) {
 	logger.Info("DevWorkspace failed to start: " + msg)
 	status.phase = devworkspacePhaseFailing
-	status.setConditionTrue(dw.DevWorkspaceFailedStart, msg)
+	status.setConditionTrue(dw.DevWorkspaceFailedStart, metrics.WorkspaceEngineFailurePrefix+msg)
```

Comment:
I'm not clear on how this works -- if we're always setting the prefix when a workspace fails (and all workspace failures come through this function at some point), it seems redundant. In which cases do we enter `determinateFailureReason` without having this prefix set?

Reply:
To clarify this a little, currently the general flow is
```
[Do something in controller]
          V
[Detect a failure state]
          V
[Set the DevWorkspaceFailed condition to true]
          V
[Set the .status.message according to the message in failure condition]
          V
[Parse this message in .status.message to distinguish failure reason]
```

It looks like what we're doing in this PR is adding a prefix to the .status.message field, but the same effect could be achieved by getting information from the failed condition.

Reply:
Yeah, I think every failure has to go through failWorkspace(). The flow is a little confusing because failure happens over 3+ reconciles:
1. [First reconcile](https://github.com/devfile/devworkspace-operator/blob/45f64af451d429ada28a96a78a5b535b3636eeec/controllers/workspace/devworkspace_controller.go#L492-L500): hit error, go through `failWorkspace` to set `Failing` status, return
2. [Second reconcile](https://github.com/devfile/devworkspace-operator/blob/45f64af451d429ada28a96a78a5b535b3636eeec/controllers/workspace/devworkspace_controller.go#L140-L159): hit early condition to check for `Failing` status, update workspace to be stopped
3. [Third reconcile](https://github.com/devfile/devworkspace-operator/blob/45f64af451d429ada28a96a78a5b535b3636eeec/controllers/workspace/devworkspace_controller.go#L161-L166): hit the `stopWorkspace` check for stopped workspaces